### ISSUE TITLE
No maps in filesystem with BPF 1.0 and xdp

### ISSUE THREAD

**Issue by tjcw** [xdp-project/xdp-tutorial#312]
I am trying to write a test case which accesses eBPF maps from a user-space application. This was available in the pre-release BPF (I am modelled on advanced03-AF_XDP) but does not seem to be available in BPF 1.0 . Have the maps moved, or what do I need to do to access them from user-space ?

**Comment by tohojo** [comment]
Chris Ward ***@***.***> writes:

> I am trying to write a test case which accesses eBPF maps from a
> user-space application. This was available in the pre-release BPF (I
> am modelled on advanced03-AF_XDP) but does not seem to be available in
> BPF 1.0 . Have the maps moved, or what do I need to do to access them
> from user-space ?

You mean the bit where maps are pinned into bpffs (under /sys/fs/bpf)?
That feature still exists in libbpf 1.0; you can do it one of two ways:
Manually (using bpf_map__pin()) in your setup code, or automatically by
adding a pinning attribute to the map definition as '__uint(pinning,
LIBBPF_PIN_BY_NAME);'

If you use the automatic feature, libbpf will pin maps under
/sys/fs/bpf/<map_name> and will also try to reuse a map if you later
(re-)load the BPF program and the map pin already exists...

**Comment by tjcw** [comment]
Yes, specifying LIBBPF_PIN_BY_NAME worked. With the earlier BPF the maps were going into /sys/fs/bpf/<device_name>/<map_name> , i.e. one more directory level. Is the change in directory expected ?

**Comment by tohojo** [comment]
Chris Ward ***@***.***> writes:

> Yes, specifying LIBBPF_PIN_BY_NAME worked. With the earlier BPF the
> maps were going into /sys/fs/bpf/<device_name>/<map_name> , i.e. one
> more directory level. Is the change in directory expected ?

You can supply a root path in your loader code by setting the
'pin_root_path' member of 'bpf_object_open_opts'. If you don't set
anything, libbpf defaults to /sys/fs/bpf
